interview-questions.md

```diff
@@ -5,50 +5,81 @@ Answer the following questions. First, without external resources. Challenge you
 1. What is object destructuring?
 
   Your answer:
-
+  To take an object and turn it into smaller, easier to handle ( and read with our human optic receptors) variables for later use. 
+  
   Researched answer:
-
-
+  "In JavaScript, destructuring is when you decompose the properties of an object or the indexes of an array to separate them to create specific variables. This does not mean that these separated objects or arrays can never be used again in the program."
 
 2. What are the three arguments that .map() can accept?
 
   Your answer:
+  Value, index and the array its mapping. 
 
   Researched answer:
-
+  "The value of the element, the index of the element, and the array object being mapped. If a thisArg parameter is provided, it will be used as callback's this value. Otherwise, the value undefined will be used as its this value"
 
 
 3. What is the difference between a class and an object?
 
   Your answer:
+  A class is kind of like a template child to the parent object, allowing it to define the properties of the parent.
 
   Researched answer:
-
+  "Classes and objects from the essential part of Object-oriented programming, where a class can be considered as a construct that encapsulates a group of variables and methods; whereas, an object acts as member or instance of that class."
 
 
 4. What is the difference between a HTML div and a span?
  
   Your answer:
+  Generally, a div tag is used for layout placement and containing large blocks of data. 
+  Span is used for smaller data, much like a paragraph tag or h1 that will only hold a small snippit of data.
 
   Researched answer:
+  "The difference between span and div is that a span element is in-line and usually used for a small chunk of HTML inside a line (such as inside a paragraph) whereas a div (division) element is block-line (which is basically equivalent to having a line-break before and after it) and used to group larger chunks of code."
 
 5. What is React.Fragment and why would you want to use it?
   
   Your answer:
+  Honestly, I'm drawing a blank. I dont think we covered it a whole lot and ontop of that, I dont feel secure in my React knowledge at all. Error 500. Proceeding to reteach myself in my research and extended study of React. 
   
   Researched answer:
+  "A common pattern in React is for a component to return multiple elements. Fragments let you group a list of children without adding extra nodes to the DOM."
+  While my researched study on this produced the above answer straight from React's website, I don't feel like this is really taking hold with me, but thats due to my unease with React. I'm currently taking an extra Udemy and a Codecademy course on React to suppliment my study.  
```

Comment:
and it's totally normal and ok to not have a solid grasp on react yet. react is kind of weird to learn linearly so it's ok! don't be too stressed :)